driver/normalizer/annotation.go

```diff
@@ -1,34 +1,354 @@
+	AnnotateType("CPPASTBreakStatement", nil, role.Break, role.Statement),
+	AnnotateType("CPPASTContinueStatement", nil, role.Continue, role.Statement),
+	AnnotateType("CPPASTLabelStatement", nil, role.Name, role.Incomplete),
+	AnnotateType("CPPASTSimpleDeclaration", nil, role.Declaration, role.Statement),
+	AnnotateType("CPPASTDeclarationStatement", nil, role.Declaration, role.Statement),
+	AnnotateType("CPPASTFieldReference", nil, role.Qualified, role.Expression),
+	AnnotateType("CPPASTBaseSpecifier", nil, role.Type, role.Declaration, role.Base),
+	AnnotateType("CPPASTNamedTypeSpecifier", nil, role.Type, role.Instance),
+	AnnotateType("CPPASTProblemStatement", nil, role.Incomplete),
+	AnnotateType("CPPASTUsingDirective", nil, role.Scope, role.Alias),
+	AnnotateType("CPPASTNewExpression", nil, role.Instance, role.Value),
+	AnnotateType("CPPASTTypeId", nil, role.Type),
+	AnnotateType("CPPASTTemplateDeclaration", nil, role.Type, role.Declaration, role.Incomplete),
+	AnnotateType("CPPASTSimpleTypeTemplateParameter", nil, role.Type, role.Declaration, role.Argument, role.Incomplete),
+	AnnotateType("CPPASTTemplateId", nil, role.Type, role.Incomplete),
+	AnnotateType("CPPASTDeleteExpression", nil, role.Call, role.Expression, role.Incomplete),
+	AnnotateType("CPPASTInitializerList", nil, role.Initialization, role.List),
+	AnnotateType("CPPASTCastExpression", nil, role.Expression, role.Incomplete),
+
+	AnnotateTypeCustom("CPPASTUnaryExpression",
+		FieldRoles{
+			"operator": {Op: Var("operator")},
+		},
+		LookupArrOpVar("operator", unaryExprRoles)),
+
+	AnnotateTypeCustom("CPPASTSimpleDeclSpecifier",
+		FieldRoles {
+			"Type": {Rename: uast.KeyToken, Op: Var("type")},
+		},
+		LookupArrOpVar("type", typeRoles),
+	),
+
+	AnnotateType("CPPASTASMDeclaration", FieldRoles{
+		"Assembly": {Rename: uast.KeyToken},
+	}, role.Declaration, role.Block, role.Incomplete),
+
+	AnnotateType("CPPASTLinkageSpecification", FieldRoles{
+		"Literal": {Rename: uast.KeyToken},
+	}, role.Declaration, role.Block, role.Incomplete),
+
+	AnnotateTypeCustom("CPPASTLiteralExpression",
+		FieldRoles {
+			"LiteralValue": {Rename: uast.KeyToken},
+			"kind": {Op: Var("kind")},
+		},
+		LookupArrOpVar("kind", litExprRoles),
+	),
+
+	AnnotateType("CPPASTCompoundStatement", nil, role.Body),
+	AnnotateType("CPPASTDeclarator", FieldRoles{
+		"Name": {Rename: uast.KeyToken},
+	}, role.Declaration, role.Variable, role.Name),
+	AnnotateType("CPPASTDeclarator", nil, role.Declaration, role.Variable, role.Name),
+
+	AnnotateType("CPPASTFunctionDefinition", ObjRoles{
+		"Prop_Body": {role.Function, role.Declaration, role.Body},
+		"Prop_DeclSpecifier": {role.Function, role.Declaration, role.Return, role.Type},
+	}, role.Function, role.Declaration),
+
+	AnnotateType("CPPASTFunctionDeclarator", FieldRoles{
+		"Prop_Name": {Roles: role.Roles{role.Function, role.Declaration, role.Name}},
+		// SDK? FIXME: adding "Opt: true" fails since Arrays can't be optional, but without it the annotation won't
```

Comment:
Yes, it might be a missed functionality in SDK, but since everything works with two rules, can you please mark it as `TODO` and not `FIXME`?

Reply:
Done!